Submitted By: Ken Moffat (ken at linuxfromscratch dot org)
Origin: fedora
        Rediffed against 4.1.0 by Chris Staub
	Rediffed against 4.2.0 by Joe Ciccone
Date: 2005-11-22
Initial package version: 4.0.2
Description: In a native ppc64 build, this fixes a linkage failure while
installing a 32-bit nof/ (no -hardware- floating point) library, and also gets
rid of 1500 failures in the g++ testsuite.  From fedora gcc-4.0.1, rediffed to
apply with -p1.

diff -Naur gcc-4.2.0.orig/gcc/config/rs6000/t-linux64 gcc-4.2.0/gcc/config/rs6000/t-linux64
--- gcc-4.2.0.orig/gcc/config/rs6000/t-linux64	2007-05-16 19:21:19.000000000 -0400
+++ gcc-4.2.0/gcc/config/rs6000/t-linux64	2007-05-16 19:51:48.000000000 -0400
@@ -7,13 +7,13 @@
 
 TARGET_LIBGCC2_CFLAGS += -mno-minimal-toc
 
-MULTILIB_OPTIONS        = m64/m32 msoft-float
-MULTILIB_DIRNAMES       = 64 32 nof
+MULTILIB_OPTIONS        = m64/m32
+MULTILIB_DIRNAMES       = 64 32
 MULTILIB_EXTRA_OPTS     = fPIC mstrict-align
-MULTILIB_EXCEPTIONS     = m64/msoft-float
-MULTILIB_EXCLUSIONS     = m64/!m32/msoft-float
-MULTILIB_OSDIRNAMES	= ../lib64 ../lib nof
-MULTILIB_MATCHES        = $(MULTILIB_MATCHES_FLOAT)
+MULTILIB_EXCEPTIONS     =
+MULTILIB_EXCLUSIONS     =
+MULTILIB_OSDIRNAMES	= ../lib64 ../lib
+MULTILIB_MATCHES        =
 
 softfp_wrap_start := '\#ifndef __powerpc64__'
 softfp_wrap_end := '\#endif'
diff -Naur gcc-4.2.0.orig/gcc/config/rs6000/t-linux64.orig gcc-4.2.0/gcc/config/rs6000/t-linux64.orig
--- gcc-4.2.0.orig/gcc/config/rs6000/t-linux64.orig	1969-12-31 19:00:00.000000000 -0500
+++ gcc-4.2.0/gcc/config/rs6000/t-linux64.orig	2007-05-16 19:51:40.000000000 -0400
@@ -0,0 +1,19 @@
+
+#rs6000/t-linux64
+
+LIB2FUNCS_EXTRA += tramp.S $(srcdir)/config/rs6000/ppc64-fp.c \
+	$(srcdir)/config/rs6000/darwin-ldouble.c
+LIB2FUNCS_EXTRA := $(sort $(LIB2FUNCS_EXTRA))
+
+TARGET_LIBGCC2_CFLAGS += -mno-minimal-toc
+
+MULTILIB_OPTIONS        = m64/m32 msoft-float
+MULTILIB_DIRNAMES       = 64 32 nof
+MULTILIB_EXTRA_OPTS     = fPIC mstrict-align
+MULTILIB_EXCEPTIONS     = m64/msoft-float
+MULTILIB_EXCLUSIONS     = m64/!m32/msoft-float
+MULTILIB_OSDIRNAMES	= ../lib64 ../lib nof
+MULTILIB_MATCHES        = $(MULTILIB_MATCHES_FLOAT)
+
+softfp_wrap_start := '\#ifndef __powerpc64__'
+softfp_wrap_end := '\#endif'
